src/west/app/main.py

```diff
@@ -594,8 +606,14 @@ def setup_west_logging(self, verbosity):
             logger.setLevel(logging.DEBUG)
         elif verbosity == 1:
             logger.setLevel(logging.INFO)
-        else:
+        elif verbosity == 0:
             logger.setLevel(logging.WARNING)
+        elif verbosity == -1:
+            logger.setLevel(logging.ERROR)
+        elif verbosity == -2:
+            logger.setLevel(logging.CRITICAL)
+        else:
+            logger.disabled = True
```

Comment:
I think it still up to the user? Scripting can rely on result codes instead of logging?

Reply:
One problem is "the user" is not always just one person / script. So there is a non-zero risk of combining `-q` (and `-v`) flags from different places and "accidentally" killing the logs.

I have been burned many times by result codes stopping automation with no error message or a very confusing one, or one drowned by further logs because of concurrency. Result codes are critical but they're not enough: decent logs are a must too.

But conversely, what is the use case for turning off CRITICAL logs? We probably don't even any yet... who could find this valuable? You can always redirect logs to /dev/null or to some never read temporary file if even CRITICAL is too much for you.